cmake/modules/FindCGAL.cmake

```diff
@@ -31,29 +31,36 @@ ENDIF()
 IF(DEAL_II_HAVE_CXX17)
   # temporarily disable ${CMAKE_SOURCE_DIR}/cmake/modules for module lookup
   LIST(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
-  FIND_PACKAGE(CGAL 5.0)
+  SET(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE ON)
+  FIND_PACKAGE(CGAL)
+  # Check version manually. Older binary distros don't do this properly.
+  IF(CGAL_MAJOR_VERSION LESS 5)
+    SET(CGAL_FOUND FALSE)
+    SET(CGAL_INCLUDE_DIRS "-NOTFOUND")
+    SET(CGAL_LIBRARIES "-NOTFOUND")
+    MESSAGE(STATUS "CGAL wrappers require CGAL version 5 and above.")
+  ENDIF()
   LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
-  IF(NOT CGAL_HEADER_ONLY)
-    SET(CGAL_LIBS "-NOTFOUND")
-    MESSAGE(STATUS "CGAL wrappers require header only version of CGAL.")
-  ELSE()
-    SET(CGAL_LIBS CGAL::CGAL)
-  ENDIF(CGAL_HEADER_ONLY)
-ELSE(DEAL_II_HAVE_CXX17)
+  IF(CGAL_FOUND)
+    GET_TARGET_PROPERTY(CGAL_LIBRARIES CGAL::CGAL INTERFACE_LINK_LIBRARIES)
+    # Make sure we dont' pass Boost::Boost over to deal.II.
+    LIST(FILTER CGAL_LIBRARIES EXCLUDE REGEX "::")
```

Comment:
This might not work with all configurations since I think this will also filter out Eigen3. I think GMP and MPFR are okay - those are not using CMake namespaces on my machine.